### PR TITLE
delete share confirmation (stable29)

### PR DESCRIPTION
**Pull request by printminion-co**
Added confirmation dialog analogue to file (>5 files) deletion.

 

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files_sharing/src/mixins/SharesMixin.js [hunk 1]
```diff
@@ -275,10 +276,46 @@ export default {
...
 		},
 
+		/**
+		 * Display delete share confirmation dialog
+		 * @returns {Promise<boolean>}
+		 */
+		async askDeleteConfirmation() {
+			let confirmed = false;
+			return (new DialogBuilder())
+				.setName(t('files_sharing', 'You are about to delete share'))
```
Comment (bromiesTM) [inline apps/files_sharing/src/mixins/SharesMixin.js:286]: typo: 
```suggestion
				.setName(t('files_sharing', 'You are about to delete a share'))
```

**Review by bromiesTM — CHANGES_REQUESTED** [review]
We should try to make this upstreamable 
Also please make sure to add yourself to the license header and ensure the compilation works properly

**Comment by printminion-co** [comment]
changed text in order to be similar to "delete files" confirmation dialog
Selection_20240806-001

**Review by thlehmann-ionos — APPROVED** [review]
Mostly OK for me, a couple of minor notes though.

- [x] Looks good to me (cancel does not delete, confirm deletes) :green_heart: 
- [x] Unit tests OK :heavy_check_mark:  
- [x] Linter OK :heavy_check_mark: 
- [x] Code style OK if one review note is addressed (semi-colon) :heavy_check_mark: 
- [x] Cypress tests don't seem to test or use this: OK   :heavy_check_mark: 

1. I'd like to see the semi-colon removed, the superfluous parens removed and the async/await vs. Promise mixture changed
2. I'm undecided on the action button type and leave it up to you since there's no clear guide (you may want to mention this when upstreaming this as a PR)

**Pull request merged**